src/core/ext/filters/http/client/http_client_filter.cc

```diff
@@ -577,17 +570,14 @@ static grpc_error_handle http_client_init_channel_elem(
   chand->static_scheme = scheme_from_args(args->channel_args);
   chand->max_payload_size_for_get =
       max_payload_size_from_args(args->channel_args);
-  chand->user_agent = grpc_mdelem_from_slices(
-      GRPC_MDSTR_USER_AGENT,
-      user_agent_from_args(args->channel_args,
-                           args->optional_transport->vtable->name));
+  new (&chand->user_agent) grpc_core::Slice(user_agent_from_args(
```

Comment:
Instead of instantiating just this one field, we should probably call the ctor for the entire `channel_data` struct, since we're calling its dtor in `http_client_destroy_channel_elem()`.  We can move the field initialization into the ctor.

Reply:
(Didn't see a response on this one; pinging just in case you missed it.)

Reply:
(didn't miss it, didn't get to it yet!)

Reply:
And done.